src/components/core/geoChart/geoChartColor.ts

```diff
@@ -36,30 +37,27 @@ export function getColorIndexInPalette(value: number, min: number, max: number):
     return DEFAULT_COLOR_INDEX_IN_PALETTE;
 }
 
-export function getColorPaletteMapping(segmentItems: string[] = []): IObjectMapping {
-    if (!segmentItems.length) {
-        return {
-            [DEFAULT_SEGMENT_ITEM]: getColorPalette(DEFAULT_COLORS[0]),
-        };
-    }
-
-    const defaultColorsNumber = DEFAULT_COLORS.length;
-    const colorPalettes = segmentItems.reduce(
-        (result: string[][], _item: string, index: number): string[][] => {
-            const colorPalette =
-                index < defaultColorsNumber
-                    ? getColorPalette(DEFAULT_COLORS[index])
-                    : result[index - defaultColorsNumber]; // colors loop every DEFAULT_COLORS.length
-            return [...result, colorPalette];
+export function getColorPaletteMapping(colorStrategy: IColorStrategy): IObjectMapping {
+    const colorAssignment: IColorAssignment[] = colorStrategy.getColorAssignment();
+    return colorAssignment.reduce(
+        (result: IObjectMapping, item: IColorAssignment, index: number): IObjectMapping => {
+            const color = colorStrategy.getColorByIndex(index);
+            const colorPalette = getColorPalette(color);
+            // color follow Location
+            if (isMappingHeaderAttribute(item.headerItem)) {
+                return {
+                    [DEFAULT_SEGMENT_ITEM]: colorPalette,
+                };
+            }
+            // color folow SegmentBy
+            const name: string = isMappingHeaderAttributeItem(item.headerItem)
+                ? item.headerItem.attributeHeaderItem.name
+                : DEFAULT_SEGMENT_ITEM;
```

Comment:
I guess `name` always be `DEFAULT_SEGMENT_ITEM ` at this point

Reply:
it should return name of element of attribute. eg: 
```
{
     headerItem: { 
         attributeHeaderItem: {
                name: "",
                uri: ""
         }
     }
}
```